src/Agent.Sdk/Knob.cs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Reflection;
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+namespace Agent.Sdk
+{
+    public enum KnobSource
+    {
+        BuiltinDefault,
+        EnvironmentVariable,
+        RuntimeVariable,
+    };
+
+    public class KnobValue
+    {
+        public KnobSource Source { get;  private set;}
+        public string Which { get; private set; }
+        private string _value;
+        public KnobValue(string value, KnobSource source, string which=null)
+        {
+            _value = value;
+            Source = source;
+            Which = which;
+        }
+
+        public string AsString()
+        {
+            return _value;
+        }
+
+        public bool AsBoolean()
+        {
+            return StringUtil.ConvertToBoolean(_value);
+        }
+    }
+
+    public class AgentKnobs : KnobPanel
+    {
+        public static readonly IKnob UseNode10 = new Knob()
+        {
+            Name="UseNode10",
+            EnvironmentVariableNames=new List<string>{ "AGENT_USE_NODE10" },
+            RuntimeVariableNames=new List<string>{ "AGENT_USE_NODE10" },
+            Description="Forces the agent to use Node 10 handler for all Node-based tasks",
+            DefaultValue="false"
+        };
+    }
+
+    public class KnobPanel
+    {
+        public class Knob : IKnob
```

Comment:
Is this class supposed to be nested?

Reply:
Yes, in order for KnobPanel to maintain a private list of instances of Knob, it has to be nested. Since I am still fairly new to C#,  there may be a better way to do this.